File: Easy/palindromeNumber.cpp
// Problem 9 :p

class Solution {
public:
  bool isPalindrome(int x) {
    long long reverse = 0;
    long long temp = x;
    while (temp != 0){
      reverse = (reverse * 10) + (temp % 10);
      temp = temp / 10;
    }
    
    if (reverse == x && x >= 0){
      return true;
    } else {
      return false;
    }
  }
};

File: Easy/palindromeNumber.rs
// problem 9 but rusty

impl Solution {
  pub fn is_palindrome(x: i32) -> bool {
    let mut reverse: i32 = 0;
    let mut temp: i32 = x;
    while temp != 0{
      reverse = (reverse * 10) + (temp % 10);
      temp = temp / 10;
    }

    if reverse == x && x >= 0{
      true
    } else {
      false
    }
  }
}